app/pages/project/instances/instance/tabs/ConnectTab.tsx

```diff
@@ -6,33 +6,102 @@
  * Copyright Oxide Computer Company
  */
 
-import { Link } from 'react-router-dom'
+import type { ReactNode } from 'react'
+import { Link, type LoaderFunctionArgs } from 'react-router-dom'
 
+import { apiQueryClient, useApiQuery } from '~/api'
 import { EquivalentCliCommand } from '~/components/EquivalentCliCommand'
-import { useInstanceSelector } from '~/hooks'
+import { getInstanceSelector, useInstanceSelector, useProjectSelector } from '~/hooks'
 import { buttonStyle } from '~/ui/lib/Button'
 import { SettingsGroup } from '~/ui/lib/SettingsGroup'
 import { cliCmd } from '~/util/cli-cmd'
 import { pb } from '~/util/path-builder'
 
+ConnectTab.loader = async ({ params }: LoaderFunctionArgs) => {
+  const { project, instance } = getInstanceSelector(params)
+  await apiQueryClient.prefetchQuery('instanceExternalIpList', {
+    path: { instance },
+    query: { project },
+  })
+  return null
+}
+
+const InlineCode = ({ children }: { children: ReactNode }) => (
+  <code className="h-4 whitespace-nowrap rounded-sm px-[3px] py-[1px] !lowercase text-mono-sm text-secondary bg-secondary">
+    {children}
+  </code>
+)
+
 export function ConnectTab() {
   const { project, instance } = useInstanceSelector()
+  const { data: externalIps } = useApiQuery('instanceExternalIpList', {
+    path: { instance },
+    query: { project },
+  })
+  const newFloatingIpLink = pb.floatingIpsNew(useProjectSelector())
+  const floatingIps = externalIps?.items?.filter((ip) => ip.kind === 'floating')
+  // default to a floating IP; fall back to ephemeral IP, if it exists
+  const externalIp = (floatingIps?.[0] || externalIps?.items?.[0])?.ip
+  const sshCopy = externalIp ? (
+    <>
+      <p>
+        If you specified SSH keys when you created this instance, you can connect to it
+        through an external IP: <InlineCode>ssh [username]@{externalIp}</InlineCode>
+      </p>
+      <p>
+        The <InlineCode>[username]</InlineCode> in the SSH command will depend on your
+        instance’s boot disk’s OS, but might be <InlineCode>debian</InlineCode>,{' '}
+        <InlineCode>ubuntu</InlineCode>, <InlineCode>arch</InlineCode>, etc.
```

Comment:
I haven't actually confirmed that `arch` is correct, or what we want to use here; open to alternates.